x/magpie/internal/keeper/keeper.go

```diff
@@ -32,14 +30,27 @@ func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey, coinKeeper bank.Keeper)
 // --- Posts
 // -------------
 
-func (k Keeper) getPostStoreKey(postId string) []byte {
-	return []byte(types.PostStorePrefix + postId)
+func (k Keeper) getPostStoreKey(postId types.PostId) []byte {
+	return []byte(types.PostStorePrefix + postId.String())
+}
+
+// GetLastPostId returns the last post id that has been used
+func (k Keeper) GetLastPostId(ctx sdk.Context) (id types.PostId) {
```

Comment:
It panics when `MsgCreatePost` create a new `Post` object and getting the last id.

```
"raw_log":"{\"codespace\":\"sdk\",\"code\":1,\"message\":\"recovered: unmarshal to types.PostId failed after 0 bytes (buffer too small): \\nstack:\\ngoroutine 99 [running]:\\nruntime/debug.Stack(0xc000b7d9b0, 0x4bd99e0, 0xc0003b0210)\\n\\t/usr/local/Cellar/go/1.13/libexec/src/runtime/debug/stack.go:24 +0x9d\\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).runTx.func1(0xc000b80180, 0xc000b808b0, 0xc000b81df8)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.37.3/baseapp/baseapp.go:886 +0x90\\npanic(0x4bd99e0, 0xc0003b0210)\\n\\t/usr/local/Cellar/go/1.13/libexec/src/runtime/panic.go:679 +0x1b2\\ngithub.com/tendermint/go-amino.(*Codec).MustUnmarshalBinaryBare(...)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/go-amino@v0.15.0/amino.go:358\\ngithub.com/kwunyeung/desmos/x/magpie/internal/keeper.Keeper.GetLastPostId(0x51193a0, 0xc000c240e0, 0x50f77e0, 0xc000c710e0, 0xc0004dcc40, 0x51061e0, 0xc0000cc008, 0x5116a20, 0xc000cc7580, 0xa, ...)\\n\\t/Users/kwunyueng/Projects/desmos/x/magpie/internal/keeper/keeper.go:40 +0x1aa\\ngithub.com/kwunyeung/desmos/x/magpie/internal/keeper.handleMsgCreatePost(0x51061e0, 0xc0000cc008, 0x5116a20, 0xc000cc7580, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\\n\\t/Users/kwunyueng/Projects/desmos/x/magpie/internal/keeper/handler.go:37 +0xab\\ngithub.com/kwunyeung/desmos/x/magpie/internal/keeper.NewHandler.func1(0x51061e0, 0xc0000cc008, 0x5116a20, 0xc000cc7580, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\\n\\t/Users/kwunyueng/Projects/desmos/x/magpie/internal/keeper/handler.go:19 +0x3ab\\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).runMsgs(0xc000c7e120, 0x51061e0, 0xc0000cc008, 0x5116a20, 0xc000cc7580, 0xa, 0x0, 0x0, 0x0, 0x0, ...)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.37.3/baseapp/baseapp.go:787 +0xbce\\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).runTx(0xc000c7e120, 0x2, 0xc000169340, 0xd6, 0xd6, 0x50f95e0, 0xc0022e03c0, 0x0, 0x0, 0x0, ...)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.37.3/baseapp/baseapp.go:955 +0x3e4\\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).DeliverTx(0xc000c7e120, 0xc000169340, 0xd6, 0xd6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.37.3/baseapp/baseapp.go:716 +0x2c1\\ngithub.com/tendermint/tendermint/abci/client.(*localClient).DeliverTxAsync(0xc0003af380, 0xc000169340, 0xd6, 0xd6, 0x0, 0x0, 0x0, 0x0, 0x0)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/abci/client/local_client.go:88 +0xed\\ngithub.com/tendermint/tendermint/proxy.(*appConnConsensus).DeliverTxAsync(0xc0000f0480, 0xc000169340, 0xd6, 0xd6, 0x0, 0x0, 0x0, 0x0, 0x0)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/proxy/app_conn.go:73 +0x5e\\ngithub.com/tendermint/tendermint/state.execBlockOnProxyApp(0x5106b60, 0xc00097dd80, 0x51100a0, 0xc0000f0480, 0xc00000d4a0, 0x5118720, 0xc0000c0670, 0x6, 0xc000c60c30, 0x6)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/state/execution.go:286 +0x613\\ngithub.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(0xc0008c8380, 0xa, 0x0, 0xc000c60c20, 0x6, 0xc000c60c30, 0x6, 0x3, 0x0, 0xc000cc3440, ...)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/state/execution.go:124 +0x17a\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).finalizeCommit(0xc000ab4000, 0x4)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:1342 +0x90a\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).tryFinalizeCommit(0xc000ab4000, 0x4)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:1273 +0x383\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).enterCommit.func1(0xc000ab4000, 0x0, 0x4)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:1218 +0x90\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).enterCommit(0xc000ab4000, 0x4, 0x0)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:1250 +0x61a\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).addVote(0xc000ab4000, 0xc000afd4a0, 0x0, 0x0, 0xc000b83ae8, 0x4030a21, 0x4f4e1a0)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:1678 +0xa39\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).tryAddVote(0xc000ab4000, 0xc000afd4a0, 0x0, 0x0, 0x8, 0xc000b83af8, 0x40b0c8d)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:1521 +0x59\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).handleMsg(0xc000ab4000, 0x50e8260, 0xc0000c4c48, 0x0, 0x0)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:688 +0x252\\ngithub.com/tendermint/tendermint/consensus.(*ConsensusState).receiveRoutine(0xc000ab4000, 0x0)\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:646 +0x56a\\ncreated by github.com/tendermint/tendermint/consensus.(*ConsensusState).OnStart\\n\\t/Users/kwunyueng/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/consensus/state.go:334 +0x13a\\n\"}"
```

Reply:
Thank you for pointing this out. Fixed it and added some tests so that it can no longer verify in the future.